finetune/saver.py

```diff
@@ -129,21 +144,48 @@ def load(self, path):
 
     def get_scaffold_init_fn(self):
         
-        def init_fn(scaffold, session):
+        def init_fn(scaffold, session, model_portion=None, refresh_base_model = False):
+            self.var_val = []
             if self.variables is not None:
                 variables_sv = self.variables
             else:
                 variables_sv = dict()
             all_vars = tf.global_variables()
-            self.var_val = []
-            for var in all_vars:
-                for saved_var_name, saved_var in itertools.chain(variables_sv.items(), self.fallback.items()):
-                    if saved_var_name == var.name:
+            if model_portion != 'entire_model': #we must be loading in the case of two separate estimators
+                reset_adapters = False
+                assert model_portion in ['featurizer','target','whole_featurizer'], "Must be using separate estimators if loading before graph creation"
+                base = [v for v in all_vars if 'target' not in v.name]
+                if model_portion == 'whole_featurizer':
+                    to_load = base
+                    adapters = [v for v in base if 'adapter' in v.name]
+                    reset_adapters = True
+                elif model_portion == 'featurizer':
+                    norm_variable_scopes = ['b:0', 'g:0']
+                    to_load = base if refresh_base_model else [v for v in base if 'adapter' in v.name or v.name[-3:] in norm_variable_scopes]
+                elif model_portion == 'target':
+                    to_load = [v for v in all_vars if 'target' in v.name]
+                for var in to_load:
```

Comment:
This code pretty much duplicates the code in the `else:` block below (and adds the reset_adapters conditional logic).  They could be merged.